File: SlidingWindow/python/leet643.py
def findMaxAverage(self, nums, k):
    """
    :type nums: List[int]
    :type k: int
    :rtype: float
    """
    i = 0
    j = 0 
    m = 0
    max = -10.0
    avg = 0.0
        
    while(j<len(nums)):
        if(j==(i+k)):
            if(max<avg/k):
                max = avg/k
            avg = 0.0
            i+=1
            j= i
        else:
            avg+=nums[j]
            j+=1
    if(j == (i+k)):
        if(max<avg/k):
            max = avg/k
    return max

File: SlidingWindow/cpp/leet643.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double sum = 0.0; 
        double ans = 0.0;
        
        for(int i = 0; i<k; ++i){
            sum+=nums[i];
        }
        sum = sum;
        ans = sum; 
        
        for(int j = k; j<nums.size(); ++j) {
            sum = sum + nums[j] - nums[j-k];
            ans = std::max(ans, sum);
        }
        return ans/k;
    }
};